libraries/SBU/extras/SBUBoot/SBUBoot.ino

```diff
@@ -0,0 +1,102 @@
+/*
+  Copyright (c) 2020 Arduino LLC.  All right reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+  See the GNU Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include <FlashStorage.h>
+#include <MKRNB.h>
+
+#define SBU_START    0x2000
+#define SBU_SIZE     0x8000
+
+#define SKETCH_START (uint32_t*)(SBU_START + SBU_SIZE)
+
+static constexpr char UPDATE_FILE_NAME[] = "UPDATE.BIN";
+static constexpr char CHECK_FILE_NAME[] = "UPDATE.OK";
+
+FlashClass mcu_flash;
+
+NBFileUtils  fileUtils(true);
+
+extern "C" void __libc_init_array(void);
+
+int main()
+{
+  init();
+
+  __libc_init_array();
+
+  delay(1);
+
+  constexpr size_t blockSize = 512;
+  fileUtils.begin();
+
+  bool update_success = false;
+
+  // Try to update only if update file
+  // has been download successfully.
+
+  if (fileUtils.existFile(CHECK_FILE_NAME)) {
+    uint32_t updateSize = fileUtils.listFile(UPDATE_FILE_NAME);
+    uint32_t tot_bytes = 0;
+    uint32_t read_bytes = 0;
+
+    if (updateSize > SBU_SIZE) {
+      updateSize = updateSize - SBU_SIZE - SBU_START;
```

Comment:
Good Morning @giulcioffi ☕ 👋 I don't think this is quite right. The reason for this is that the first 0x2000 bytes of the MCU flash are occupied by the bootloader (the default bootloader, not the SBU) but those 0x2000 bytes of default bootloader are not included in the binary generated when compiling your Arduino sketch. Therefore I think it very likely that with the code like this you're skipping the last 0x2000 bytes of the actual application. 
```diff
-updateSize = updateSize - SBU_SIZE - SBU_START;
+updateSize = updateSize - SBU_SIZE;
```
For reference compare to [SSU](https://github.com/arduino/ArduinoCore-samd/blob/master/libraries/SSU/extras/SSUBoot/SSUBoot.ino#L110).